python/pyarrow/_compute.pyx

```diff
@@ -513,6 +514,13 @@ def function_registry():
     return _global_func_registry
 
 
+def make_function_registry():
+    up_registry = MakeFunctionRegistry()
+    c_registry = up_registry.get()
+    up_registry.release()
```

Comment:
AFAICT, this effectively creates a memory leak as there's no `unique_ptr` owning the pointer anymore.